cores/nRF5/HardwarePWM.cpp

```diff
@@ -289,10 +289,13 @@ bool HardwarePWM::takeOwnership(uintptr_t token)
 
   // This function must not be called within ISR
   taskENTER_CRITICAL();
-  _owner_token = token;
+  if ( this->_owner_token == 0 )
+  {
+    _owner_token = token;
+  }
   taskEXIT_CRITICAL();
 
-  return true;
+  return _owner_token == token;
```

Comment:
You have removed the 'atomic' for `_owner_token`.
You have not marked it volatile.
I have no idea if checking `_owner_token` outside the critical section is "safe" with your code.
You do this at your own risk.